sink/datadog.go

```diff
@@ -92,23 +96,49 @@ func NewDatadog(monitorId string, opts, tags map[string]string, httpClient *http
 				return nil, err
 			}
 			d.tr = tr
+
 		case "metric-prefix":
 			if v == "" {
 				return nil, fmt.Errorf("datadog sink metric-prefix is empty string; value required when option is specified")
 			}
 			d.prefix = v
 
+		case "dogstatsd-host":
+			d.dogStatsDHost = v
+
 		default:
 			return nil, fmt.Errorf("invalid option: %s", k)
 		}
 	}
 
-	if d.apiKeyAuth == "" {
-		return nil, fmt.Errorf("datadog sink required either api-key-auth or api-key-auth-file")
+	if d.dogStatsDHost != "" {
+		d.dogStatsD = true
 	}
 
-	if d.appKeyAuth == "" {
-		return nil, fmt.Errorf("datadog sink required either app-key-auth or app-key-auth-file")
+	// if DogStatsD and api are both setup, return error as it will otherwise result in duplicate metrics
+	if d.dogStatsD && (d.apiKeyAuth != "" && d.appKeyAuth != "") {
+		return nil, fmt.Errorf("datadog sink requires either dogStatsD host or (api-key-auth and app-key-auth), not both at the same time")
+	}
+
+	if d.dogStatsD {
+		client, err := statsd.New(fmt.Sprintf("%s:%d", d.dogStatsDHost, DefaultDogStatsDPort))
```

Comment:
You'll need to check and append port only if needed, like https://github.com/cashapp/blip/blob/main/aws/iam_auth.go#L23